src/main/java/uk/ac/ebi/atlas/experimentpage/markergenes/HighchartsHeatmapAdapter.java

```diff
@@ -77,22 +79,28 @@ public HighchartsHeatmapAdapter(BioEntityPropertyDao bioEntityPropertyDao) {
      * The rows of the heatmap are ordered by the cell type, i.e. genes for celltype 1, 2, etc.
      * If there are no marker genes for a cell group, then no rows will be present in the data.
      */
-    public ImmutableList<ImmutableMap<String, Object>> getCellTypeMarkerGeneHeatmapData(Collection<CellTypeMarkerGene> markerGenes) {
+    public ImmutableList<ImmutableMap<String, Object>> getCellTypeMarkerGeneHeatmapData(Map<String, ImmutableSet<CellTypeMarkerGene>> markerGenesMap) {
         // Whether the comparison by p-value should or shouldn’t be reversed depends on the yAxis.reversed property in
         // the heatmap component. In our case it’s set to true, so lower p-value is displayed at the top without
         // reversing the comparator
-        var sortedMarkerGenes = markerGenes.stream()
-                .parallel()
-                .sorted(comparing(CellTypeMarkerGene::cellGroupValueWhereMarker).thenComparing(CellTypeMarkerGene::pValue))
-                .collect(toImmutableList());
 
-        var rows =
-                sortedMarkerGenes.stream()
+        // Convert Map to List.This will merge all top5 cell type marker genes ImmutableSets into a a single
+        // List to use existing codebase approach of HighchartsHeatmapAdapter.
+        List<CellTypeMarkerGene> allMarkerGenes = markerGenesMap.values().stream()
+                                                    .flatMap(Collection::stream)
+                                                    .collect(Collectors.toList());
+
+        var sortedMarkerGenes = allMarkerGenes.stream()
+                                .parallel()
+                                .sorted(comparing(CellTypeMarkerGene::cellGroupValueWhereMarker).thenComparing(CellTypeMarkerGene::pValue))
+                                .collect(toImmutableList());
+
+        var rows = sortedMarkerGenes.stream()
```

Comment:
You can chain these two blocks, right?

Reply:
Yes, we can. I want to get rid of this Branch, as I duplicated the same code in the [Update cluster PR](https://github.com/ebi-gene-expression-group/atlas-web-single-cell/pull/113). I will fix these comments in the Update cluster PR.